src/common/pubsub.ts

```diff
@@ -133,10 +134,35 @@ export const notifyUserReputationUpdated = async (
     reputation,
   });
 
-export const notifyAlertsUpdated = async (
+export const removeSomeProps = <T>(obj: T, excludes: (keyof T)[]): Partial<T> =>
+  Object.keys(obj).reduce((result, key) => {
+    if (excludes.some((excluding) => excluding === key)) {
+      return result;
+    }
+
+    return { ...result, [key]: obj[key] };
+  }, {});
+
+export const notifyAlertsUpdated = (
   log: EventLogger,
   alerts: ChangeObject<Alerts>,
-): Promise<void> => publishEvent(log, alertsFilterUpdatedTopic, alerts);
+): Promise<void> =>
+  publishEvent(log, alertsUpdatedTopic, removeSomeProps(alerts, ['userId']));
+
+export const excludedSettingsProps: (keyof Settings)[] = [
+  'userId',
+  'updatedAt',
+];
+
+export const notifySettingsUpdated = (
+  log: EventLogger,
+  settings: ChangeObject<Settings>,
+): Promise<void> =>
+  publishEvent(
+    log,
+    settingsUpdatedTopic,
+    removeSomeProps(settings, excludedSettingsProps),
```

Comment:
Why do you remove the `userId` and `updatedAt` props from the event?

Reply:
Was thinking it won't be too much of use for the other end. But I realized it should be the received end's concern if they will not process some of the data. Let me fix this one.